packages/expo-router/src/modal/__tests__/detents.test.ios.ts

```diff
@@ -33,3 +34,33 @@ describe(areDetentsValid, () => {
     });
   });
 });
+
+describe(isInitialDetentIndexValid, () => {
+  describe('valid initialDetentIndex', () => {
+    it.each([
+      { detents: [0, 0.5, 1], detentIndex: 1 },
+      { detents: [0.1, 0.2, 0.9], detentIndex: 2 },
+      { detents: [0.1, 0.9], detentIndex: 'last' },
```

Comment:
nit:

```suggestion
      { detents: [0.1, 0.9], detentIndex: 'last' as const},
```

This way you can remove the `// @ts-expect-error`